src/unix/signal.c

```diff
@@ -944,8 +944,12 @@ closure_function(1, 2, void, signalfd_notify,
         sig_debug("%d spurious notify\n", sfd->fd);
         return;
     }
-    blockq_wake_one_for_thread(sfd->bq, t);
-    notify_dispatch_for_thread(sfd->f.ns, EPOLLIN, t);
+
+    /* null thread on notify set release (thread dealloc) */
+    if (t) {
+        blockq_wake_one_for_thread(sfd->bq, t);
+        notify_dispatch_for_thread(sfd->f.ns, EPOLLIN, t);
```

Comment:
To keep the current behavior, notify_dispatch_for_thread() should be called even if `t==0`, otherwise any threads doing polling on a signalfd won't be notified when the signalfd is closed (all notify handlers are safe to be called with a null thread argument).

Reply:
Yes, I agree that shouldn't be conditional. Fix applied.